From d5064cf06cd75d8717848072e09b395de1de46ac Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Sun, 12 Feb 2012 18:17:52 +0800
Subject: [PATCH 198/474] edac: fix edac build error

extract the patch from mainline commit
3653ada5d3e173489b3a466305687cb5c44b2ab1

and it fixes following compile error:
drivers/edac/edac_mce_amd.c: In function 'mce_amd_init':
drivers/edac/edac_mce_amd.c:433: error: 'x86_mce_decoder_chain' undeclared (first use in this function)
drivers/edac/edac_mce_amd.c:433: error: (Each undeclared identifier is reported only once
drivers/edac/edac_mce_amd.c:433: error: for each function it appears in.)
remake[8]: *** [drivers/edac/edac_mce_amd.o] Error 1

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 drivers/edac/edac_mce_amd.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/edac/edac_mce_amd.c b/drivers/edac/edac_mce_amd.c
index 97e64bc..a30ffc5 100644
--- a/drivers/edac/edac_mce_amd.c
+++ b/drivers/edac/edac_mce_amd.c
@@ -430,7 +430,7 @@ static int __init mce_amd_init(void)
 	 */
 	if ((boot_cpu_data.x86_vendor == X86_VENDOR_AMD) &&
 	    (boot_cpu_data.x86 >= 0xf))
-		atomic_notifier_chain_register(&x86_mce_decoder_chain, &amd_mce_dec_nb);
+		mce_register_decode_chain(&amd_mce_dec_nb);
 
 	return 0;
 }
@@ -439,7 +439,7 @@ early_initcall(mce_amd_init);
 #ifdef MODULE
 static void __exit mce_amd_exit(void)
 {
-	atomic_notifier_chain_unregister(&x86_mce_decoder_chain, &amd_mce_dec_nb);
+	mce_unregister_decode_chain(&amd_mce_dec_nb);
 }
 
 MODULE_DESCRIPTION("AMD MCE decoder");
-- 
1.7.0

